crates/rattler_pypi_interop/src/types/pypi_version_types.rs

```diff
@@ -0,0 +1,94 @@
+//! This module contains types that are used to represent versions and version sets
+//! these are used by the [`resolvo`] crate to resolve dependencies.
+
+use crate::types::{Extra, NormalizedPackageName};
+use pep440_rs::Version;
+use serde::{Deserialize, Serialize};
+use std::fmt::{Display, Formatter};
+use url::Url;
+
+/// This is a wrapper around [`Version`]. It can also be a direct url to a package.
+#[derive(Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub enum PypiVersion {
```

Comment:
We could use https://docs.rs/pep508_rs/latest/pep508_rs/enum.VersionOrUrl.html for this. But is it even used?